app/views/viewlets/change-version-jsx.jsx

```diff
@@ -0,0 +1,84 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2013 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+
+YUI.add('change-version-view-jsx', function(Y) {
+
+  var ns = Y.namespace('juju.viewlets');
+
+  ns.ChangeVersionList = React.createClass({
+    // React is not MVC use utility classes and pass them into views.
+    // If it doesn't involve the UI do not put it in the component.
+    _upgradeService: function(upgradeTo) {
+      var props = this.props,
+          env = props.viewletManager.env,
+          setCharm = env.setCharm;
+      setCharm.call(env, props.model.id, upgradeTo, false);
+    },
+
+    render: function() {
+      var items = [],
+          availableVersions = this.props.model.available_versions;
+      if (availableVersions) {
+        availableVersions.forEach(function(version) {
+          items.push(<ns.ChangeVersionItem version={version} upgradeService={this._upgradeService} />)
+        }, this);
+      }
+      // Note JSX is not HTML, to set classes you use the jsdom api key
+      // className.
+      return (
+        <div>
+          <div className="unit-list-header">
+            <div className="change-version-close link">
+              <i className="sprite close-inspector-click"></i>
+            </div>
+            Change version
+          </div>
+          <ul className="unit-list">
+            {items}
+          </ul>
+        </div>
+      );
+    }
+  });
+
+  ns.ChangeVersionItem = React.createClass({
```

Comment:
Putting every component in this namespace makes it trivial to test. We can even test these components in nodejs without phantom. But this is level of division isn't necessary I only did it as such to illustrate how we would do things like the charm lists, machine lists etc. Where the line items have a little bit more functionality.